mobile/test/common/integration/test_server.cc

```diff
@@ -193,143 +193,335 @@ void TestServer::setHeadersAndData(absl::string_view header_key, absl::string_vi
+                      base_interval {
+                        seconds: 2
+                      }
+                      max_interval {
+                        seconds: 10
+                      }
+                    }
+                    dns_query_timeout {
+                      seconds: 25
+                    }
+                    typed_dns_resolver_config {
+                      name: "envoy.network.dns_resolver.getaddrinfo"
+                      typed_config {
+                        [type.googleapis.com/envoy.extensions.network.dns_resolver.getaddrinfo.v3.GetAddrInfoDnsResolverConfig] {
+                        }
+                      }
+                    }
+                    dns_min_refresh_rate {
+                      seconds: 20
+                    }
+                  }
+                }
+              }
+            }
+            http_filters {
+              name: "envoy.router"
+              typed_config {
+                [type.googleapis.com/envoy.extensions.filters.http.router.v3.Router] {
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  clusters {
+    name: "cluster_proxy"
+    connect_timeout {
+      seconds: 30
+    }
     lb_policy: CLUSTER_PROVIDED
     dns_lookup_family: ALL
-    cluster_type:
-      name: envoy.clusters.dynamic_forward_proxy
-      typed_config:
-        "@type": type.googleapis.com/envoy.extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig
-        dns_cache_config: *dns_cache_config
-layered_runtime:
-  layers:
-    - name: static_layer_0
-      static_layer:
-        envoy:
-          # This disables envoy bug stats, which are filtered out of our stats inclusion list anyway
-          # Global stats do not play well with engines with limited lifetimes
-          disallow_global_stats: true
+    cluster_type {
+      name: "envoy.clusters.dynamic_forward_proxy"
+      typed_config {
+        [type.googleapis.com/envoy.extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig] {
+          dns_cache_config {
+            name: "base_dns_cache"
+            dns_lookup_family: ALL
+            dns_refresh_rate {
+              seconds: 60
+            }
+            host_ttl {
+              seconds: 86400
+            }
+            dns_failure_refresh_rate {
+              base_interval {
+                seconds: 2
+              }
+              max_interval {
+                seconds: 10
+              }
+            }
+            dns_query_timeout {
+              seconds: 25
+            }
+            typed_dns_resolver_config {
+              name: "envoy.network.dns_resolver.getaddrinfo"
+              typed_config {
+                [type.googleapis.com/envoy.extensions.network.dns_resolver.getaddrinfo.v3.GetAddrInfoDnsResolverConfig] {
+                }
+              }
+            }
+            dns_min_refresh_rate {
+              seconds: 20
+            }
+          }
+        }
```

Comment:
seems like the diff is a lot larger here b/c of formatting changes?  I'm just going to assume the yaml is right :)